phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -322,28 +370,19 @@ private void repairIndexRows(byte[] indexRowKey, long ts, List<Cell> row) throws
                 singleRowScanner.next(row);
                 singleRowScanner.close();
                 if (row.isEmpty()) {
-                    // This means that the data row did not exist, so we need to skip this unverified row (i.e., do not
-                    // return it to the client). Just retuning empty row is sufficient to do that
-                    return;
+                    throw new DoNotRetryIOException("Could not find the newly rebuilt index row " +
```

Comment:
I am not sure for throwing exception here. If we can't build 1 index row, we will error out and skip the next ones. Would not this create more issues down the line?

Reply:
This and the previous exception are to catch "impossible" cases. Now that you asked, I want to check more time if they really cannot happen. Maybe logging error and skipping could be safer as you suggested.

Reply:
I have looked into this again. The first case can be encountered due to compaction. But this one is really an exception as we have a can filter for a specific row but we got a different row. I updated the code accordingly. Thx

Reply:
Thank you @kadirozde !